applications/OptimizationApplication/custom_utilities/container_variable_data_holder/container_variable_data_holder_base.cpp

```diff
@@ -0,0 +1,180 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   license: OptimizationApplication/license.txt
+//
+//  Main author:     Suneth Warnakulasuriya
+//
+
+// System includes
+#include <sstream>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "utilities/parallel_utilities.h"
+
+// Application includes
+
+// Include base h
+#include "container_variable_data_holder_base.h"
+
+namespace Kratos {
+
+template <class TContainerType>
+ContainerVariableDataHolderBase<TContainerType>::ContainerVariableDataHolderBase(ModelPart& rModelPart)
+    : mDataDimension(0),
+      mpModelPart(&rModelPart)
```

Comment:
```suggestion
    : mData(),
      mDataDimension(0),
      mpModelPart(&rModelPart)
```

if you're going for direct initializations, you need to explicitly initialize **all** members in the **exact order** they were defined in the class.